org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/EclipseInputFile.java

```diff
@@ -67,17 +66,21 @@ private synchronized void initFromFS() {
       fileStore = EFS.getStore(file.getLocationURI());
       File localFile = fileStore.toLocalFile(EFS.NONE, null);
       if (localFile == null) {
-        // Try to get a cached copy (for virtual file systems)
-        localFile = fileStore.toLocalFile(EFS.CACHE, null);
+        // For analyzers to properly work we should ensure the temporary file has a "correct" name, and not a generated one
+        localFile = new File(tempDirectory.toFile(), file.getProjectRelativePath().toOSString());
+        Files.createDirectories(localFile.getParentFile().toPath());
+        fileStore.copy(EFS.getStore(localFile.toURI()), EFS.OVERWRITE, null);
       }
       filePath = localFile.toPath().toAbsolutePath();
-    } catch (CoreException e) {
+    } catch (Exception e) {
       throw new IllegalStateException("Unable to find path for file " + file, e);
     }
   }
 
   @Override
   public String getPath() {
+    // TODO When we are sure all analyzers stop using this to read file content, we should return "virtual" path and not create temporary
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)